earth_enterprise/src/installer/common.sh

```diff
@@ -88,20 +106,14 @@ software_check()
     # args: $2: Ubuntu package
     # args: $3: RHEL package
 
-    if [ "$MACHINE_OS" == "$UBUNTUKEY" ] && [ ! -z "$2" ]; then
-        if [[ -z "$(dpkg --get-selections | sed s:install:: | sed -e 's:\s::g' | grep ^$2\$)" ]]; then
+    if ! is_package_installed $2 $3 ; then
+	if [ "$MACHINE_OS" == "$UBUNTUKEY" ] && [ ! -z "$2" ]; then
             echo -e "\nInstall $2 and restart the $1."
             software_check_retval=1
-        fi
-    elif { [ "$MACHINE_OS" == "$REDHATKEY" ] || [ "$MACHINE_OS" == "$CENTOSKEY" ]; } && [ ! -z "$3" ]; then
-        if [[ -z "$(rpm -qa | grep ^$3\$)" ]]; then
+	elif { [ "$MACHINE_OS" == "$REDHATKEY" ] || [ "$MACHINE_OS" == "$CENTOSKEY" ]; } && [ ! -z "$3" ]; then 
             echo -e "\nInstall $3 and restart the $1."
             software_check_retval=1
         fi
-    else
```

Comment:
I agree - it would be good to keep the else.

Reply:
Well, the reason I took it out is that the determine_os function has to be called before the software checks (to fill in MACHINE_OS) and it already checks for an unsupported OS and prints the list of supported systems.  Since the installers exit when determine_os fails it shouldn't be possible for these checks to run on an unsupported OS.

But it's less fragile to handle it here, too, in case determine_os changes, so I'll add it back in.